cockatrice/src/client/ui/widgets/visual_deck_storage/visual_deck_storage_folder_display_widget.cpp

```diff
@@ -0,0 +1,186 @@
+    auto filteredByTags = tagFilterWidget->filterDecksBySelectedTags(filteredByColorIdentity);
+    auto filteredFiles = searchWidget->filterFiles(filteredByTags, searchWidget->getSearchText());
+    */
+
+    flowWidget->clearLayout(); // Clear existing widgets in the flow layout
+
+    for (DeckPreviewWidget *deck : allDecks) {
+        flowWidget->addWidget(deck);
+    }
+
+    // emit bannerCardsRefreshed();
+}
+
+void VisualDeckStorageFolderDisplayWidget::updateVisibility()
+{
+    bool atLeastOneWidgetVisible = checkVisibility();
+    if (atLeastOneWidgetVisible) {
+        setVisible(true);
+        for (DeckPreviewWidget *display : flowWidget->findChildren<DeckPreviewWidget *>()) {
+            display->updateVisibility();
+        }
+        for (VisualDeckStorageFolderDisplayWidget *subFolder : findChildren<VisualDeckStorageFolderDisplayWidget *>()) {
+            subFolder->updateVisibility();
+        }
+    } else {
+        setVisible(false);
+    }
+}
+
+bool VisualDeckStorageFolderDisplayWidget::checkVisibility()
+{
+    bool atLeastOneWidgetVisible = false;
+    if (flowWidget) {
+        // Iterate through all DeckPreviewWidgets
+        for (DeckPreviewWidget *display : flowWidget->findChildren<DeckPreviewWidget *>()) {
+            // Get tags from each DeckPreviewWidget
+            if (display->checkVisibility()) {
+                atLeastOneWidgetVisible = true;
+            }
+        }
+    }
+    for (VisualDeckStorageFolderDisplayWidget *subFolder : findChildren<VisualDeckStorageFolderDisplayWidget *>()) {
+        if (subFolder->checkVisibility()) {
+            atLeastOneWidgetVisible = true;
+        }
+    }
+    return atLeastOneWidgetVisible;
+}
+
+void VisualDeckStorageFolderDisplayWidget::createWidgetsForFolders()
+{
+    for (const QString &dir : getAllSubFolders()) {
+        auto *display = new VisualDeckStorageFolderDisplayWidget(this, visualDeckStorageWidget, dir);
+        containerLayout->addWidget(display);
+    }
+}
+
+QStringList VisualDeckStorageFolderDisplayWidget::gatherAllTagsFromFlowWidget() const
+{
+    QStringList allTags;
+
+    if (flowWidget) {
+        // Iterate through all DeckPreviewWidgets
+        for (DeckPreviewWidget *display : flowWidget->findChildren<DeckPreviewWidget *>()) {
+            // Get tags from each DeckPreviewWidget
+            QStringList tags = display->deckLoader->getTags();
+
+            // Add tags to the list while avoiding duplicates
+            allTags.append(tags);
+        }
+    }
+
+    // Remove duplicates by calling 'removeDuplicates'
+    allTags.removeDuplicates();
+
+    return allTags;
+}
+
+QStringList VisualDeckStorageFolderDisplayWidget::getAllFiles() const
+{
+    QStringList allFiles;
+
+    // QDirIterator with QDir::Files and QDir::NoSymLinks ensures only files are listed (no directories or symlinks)
+    QDirIterator it(filePath, QDir::Files | QDir::NoSymLinks);
```

Comment:
I don't think I agree w/ no sym links on this one. People might link out their own stuff to external sources, and we don't want to ignore them inadvertently.